### PR TITLE
STM32 *.dtsi: implement `STM32_CLOCK` macro to better format values in DTSI

### LINKED ISSUE #78106: STM32 *.dtsi: configure `clocks` via macro similar to `STM32_RESET`
**Is your enhancement proposal related to a problem? Please describe.**
Currently, the `resets` property of nodes in STM32 DTSI files is filled using a macro called `STM32_RESET`.
https://github.com/zephyrproject-rtos/zephyr/blob/34982bb00b57486aeb83a6cb3bb7433791302aed/dts/arm/st/c0/stm32c0.dtsi#L177-L184

The macro's implementation is found in `include/zephyr/dt-bindings/reset/stm32-common.h`:
https://github.com/zephyrproject-rtos/zephyr/blob/34982bb00b57486aeb83a6cb3bb7433791302aed/include/zephyr/dt-bindings/reset/stm32-common.h#L20

Ignoring the bitpacking detail, this macro provides two benefits for users:
1. removes the need to type `STM32_RESET_BUS_` every time
2. accepts a bit number, which is the form used in RefMan

TL;DR: this macro makes both authoring ***and*** reading of DTSI files a much better experience.

Currently, there is unfortunately no equivalent to this macro for the `clocks` property.
To make matters worse, (nearly) all DTSI files are using raw hex values for the `clocks` properties. This makes reading and authoring of these files tedious.

**Describe the solution you'd like**

A new macro - ideally named `STM32_CLOCK` - should be created to offer the same feature for `clocks` property:
```c
/* Whitespace between 'STM32_CLOCK_BUS_##bus' and '(1 << (bit))'
 * is necessary - this macro must expand to two DTS cells.
 */
#define STM32_CLOCK(bus, bit) STM32_CLOCK_BUS_##bus  (1 << (bit))
```

> However, note that `STM32_CLOCK` is already used for domain clock (a.k.a. kernel clock) selection:
> https://github.com/zephyrproject-rtos/zephyr/blob/34982bb00b57486aeb83a6cb3bb7433791302aed/include/zephyr/dt-bindings/clock/stm32c0_clock.h#L54-L58
> There are two solutions to this issue:
> * Circumventing by using a different name (for example, `STM32_CLOCK_GATE`)
> * Renaming the existing macro beforehand (for example, `STM32_CLOCK` --> `STM32_DOMAIN_CLOCK`)
>   * This *should* not be a breaking change, as this macro is for internal use (wrapped by `xxx_SEL()` which are for public usage) 
>   * N.B.: A revamp of this macro is already under consideration; if performed, this issue may be solved as a bonus

After the introduction of this macro, the existing DTSI files would have to be modified to use it.

Taking the earlier example: (note that `1 << 14` is `0x00004000`)
```diff
 usart1: serial@40013800 { 
 	compatible = "st,stm32-usart", "st,stm32-uart"; 
 	reg = <0x40013800 0x400>; 
- 	clocks = <&rcc STM32_CLOCK_BUS_APB1_2 0x00004000>;
+ 	clocks = <&rcc STM32_CLOCK(APB1_2, 14U)>; 
 	resets = <&rctl STM32_RESET(APB1H, 14U)>; 
 	interrupts = <27 0>; 
 	status = "disabled"; 
 }; 
```

**Describe alternatives you've considered**

Without introducing any macro, all raw hex values in DTSI files could be replaced with the equivalent `(1 << X)` notation.
This would make DTSI files more readable, but still tedious to write.

### PR DESCRIPTION
**Pull request by HubertYGuan**
This PR adds a new macro in `include/zephyr/dt-bindings/clock/stm32_common_clocks.h` to replace raw hex codes used for the `clocks` property of nodes in the STM32 DTSI files. It also refactors said DTSI files to use this macro. Additionally, it renames previous macros with the same name used to set domain clocks (such as in `include/zephyr/dt-bindings/clock/stm32c0_clock.h`) to `STM32_DOMAIN_CLOCK`.

Fixes: #78106

### CODE REVIEW

**Comment by erwango** [comment]
Please rebase to solve reported conflicts and fix issues reported by CI, but otherwise thanks for addressing this change.

**Review by mathieuchopstm — COMMENTED** [review]
Haven't double-checked every value but looks good to me on the surface.

Please rework the tree to remove the merge commits, as requested by CI.

Nit: `U` suffix should not be used for integers in Device Tree files (it is silently discarded by `dtlib`), but don't bother changing all files again at this stage...

**Review by mathieuchopstm — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Comment by mathieuchopstm** [comment]
5a31325d420 seems to have broken the following such nodes:
```diff
@@ dts/arm/st/f1/stm32f105.dtsi L52 @@
-			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x06000000>;
+			clocks = <&rcc STM32_CLOCK(APB1, 26U)>;

@@ dts/arm/st/f4/stm32f405.dtsi L223 @@
-			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x06000000>;
+			clocks = <&rcc STM32_CLOCK(APB1, 26U)>;

@@ dts/arm/st/f4/stm32f412.dtsi L233 (can2) @@
-			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x06000000>;
+			clocks = <&rcc STM32_CLOCK(APB1, 26U)>;

@@ dts/arm/st/f4/stm32f446.dtsi L76 (can2) @@
-			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x06000000>;
+			clocks = <&rcc STM32_CLOCK(APB1, 26U)>;

@@ dts/arm/st/u5/stm32u5.dtsi L307 (backup_sram) @@
-			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x10020000>;
+			clocks = <&rcc STM32_CLOCK(AHB1, 28U)>;

@@ dts/arm/st/u5/stm32u595.dtsi L109 (usbotg_hs) @@
- 			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x0000c000>,
+			clocks = <&rcc STM32_CLOCK(AHB2, 15U)>,
```

Creating a new PR to fix this.
